pkg/server/conn.go

```diff
@@ -1170,20 +1176,25 @@ func (cc *clientConn) addMetrics(cmd byte, startTime time.Time, err error) {
 		return
 	}
 
+	vars := cc.getCtx().GetSessionVars()
+	resource_group := vars.ResourceGroupName
 	var counter prometheus.Counter
-	if err != nil && int(cmd) < len(server_metrics.QueryTotalCountErr) {
-		counter = server_metrics.QueryTotalCountErr[cmd]
-	} else if err == nil && int(cmd) < len(server_metrics.QueryTotalCountOk) {
-		counter = server_metrics.QueryTotalCountOk[cmd]
+	if len(resource_group) == 0 {
```

Comment:
Why check resource_group length here? If the resource group name is empty you should still add the metrics with the label value of ""

Reply:
It can't affect the performance if there is only one group.

Reply:
I mean you needn't check the group name, if the name is empty, we should still track the metrics with label value ""